DATACASS-188 - now able to supply options to table via cassandra admin otemplate

Mikhail2048 opened the pull request:
Implements #359
@mp911de, can you please check?)

mp911de commented on spring-data-cassandra/src/main/java/org/springframework/data/cassandra/core/CassandraAdminOperations.java:
```diff
@@ -46,15 +47,31 @@ public interface CassandraAdminOperations extends CassandraOperations {
 	 * @param entityClass The class whose fields determine the columns created.
 	 * @param optionsByName Table options, given by the string option name and the appropriate option value.
 	 */
+	@Deprecated(forRemoval = true)
```
We do not intent deprecating this method as a table can be created from the given `Class` using a different table name.

mp911de commented on spring-data-cassandra/src/main/java/org/springframework/data/cassandra/core/CassandraAdminTemplate.java:
```diff
@@ -107,11 +110,27 @@ public SchemaFactory getSchemaFactory() {
+				.getCreateTableSpecificationFor(entity, entity.getTableName())
+				.ifNotExists(ifNotExists);
+
+		if (!CollectionUtils.isEmpty(optionsByName)) {
```
This is actually a bugfix. Can you please split this pull request into two PR's, one for introducing an overload without accepting a table name and the other one to consider options?

mp911de commented on spring-data-cassandra/src/main/java/org/springframework/data/cassandra/core/CassandraTemplate.java:
```diff
@@ -961,7 +961,7 @@ protected <E extends CassandraMappingEvent<T>, T> void maybeEmitEvent(Supplier<E
 
 		if (null != entityCallbacks) {
-			return (T) entityCallbacks.callback(BeforeConvertCallback.class, object, tableName);
+			return entityCallbacks.callback(BeforeConvertCallback.class, object, tableName);
```
These should go into a separate commit as well to separate polishing from the actual change.

mp911de reviewed (changes requested):
There are quite some unrelated changes within a single commit. Please split at least the bugfix from the newly introduced method. The polishing changes should be part of a separate commit, can be either PR.

Mikhail2048 commented:
@mp911de I have removed new method from this PR. Also I have introduced 2 commits - one with polishing, another with actual bugfix. Can you please re-review?
 
P.S: I will introduce new method once this changes will occur in the _main_ branch

Mikhail2048 commented:
Mark @mp911de, when you will be able to rereview this PR?) I have already created a separate brunch for new overloaded method, but I am waiting for these changes to occur in main brunch to create new PR as we discussed
